client/src/components/JurisdictionAdmin/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,645 @@
+import React, { BaseSyntheticEvent, useEffect, useState } from 'react'
+import styled from 'styled-components'
+import {
+  ArrayField,
+  SubmitErrorHandler,
+  SubmitHandler,
+  useFieldArray,
+  useForm,
+  UseFormMethods,
+} from 'react-hook-form'
+import {
+  Button,
+  Classes,
+  Colors,
+  H4,
+  HTMLTable,
+  Tab,
+  Tabs,
+} from '@blueprintjs/core'
+import { ErrorMessage } from '@hookform/error-message'
+
+import useContestsJurisdictionAdmin from './useContestsJurisdictionAdmin'
+import { Confirm, useConfirm } from '../Atoms/Confirm'
+import { Detail, List, ListAndDetail, ListItem } from '../Atoms/ListAndDetail'
+import {
+  IBatch,
+  IBatchResultTallySheet,
+  useBatches,
+  useRecordBatchResults,
+} from './useBatchResults'
+import { IContest } from '../../types'
+import { sum } from '../../utils/number'
+import { useDebounce } from '../../utils/debounce'
+
+const HIDDEN_INPUT_CLASS_NAME = 'hidden-input'
+
+const BatchResultTallySheetTable = styled(HTMLTable).attrs({
+  bordered: true,
+  striped: true,
+})`
+  &.${Classes.HTML_TABLE} {
+    border: 1px solid ${Colors.LIGHT_GRAY1};
+    table-layout: fixed;
+    width: 100%;
+  }
+
+  &.${Classes.HTML_TABLE} tbody tr {
+    height: 56px;
+  }
+
+  &.${Classes.HTML_TABLE} td {
+    vertical-align: middle;
+  }
+
+  .${Classes.INPUT}.${HIDDEN_INPUT_CLASS_NAME} {
+    display: none;
+  }
+`
+
+const Actions = styled('div')`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  margin-top: 16px;
+
+  .${Classes.INPUT}.${HIDDEN_INPUT_CLASS_NAME} {
+    display: none;
+  }
+`
+
+const SheetActions = styled('div')`
+  align-items: center;
+  display: flex;
+`
+
+const ConfirmSheetRenameButton = styled(Button)`
+  margin-left: 8px;
+`
+
+const SheetRenameErrorMessage = styled('span').attrs({
+  className: Classes.TEXT_SMALL,
+})`
+  color: ${Colors.RED2};
+  margin-left: 8px;
+`
+
+const Spacer = styled('div')`
+  flex-grow: 1;
+`
+
+interface IBatchRoundDataEntryFormState {
+  resultTallySheets: IBatchResultTallySheet[]
+}
+
+const VOTE_TOTALS_TAB_ID = 'vote-totals'
+
+function defaultResultTallySheetName(index: number) {
+  return `Sheet ${index + 1}`
+}
+
+function constructEmptyResultTallySheet(index: number): IBatchResultTallySheet {
+  return {
+    name: defaultResultTallySheetName(index),
+    results: {},
+  }
+}
+
+interface IProps {
+  electionId: string
+  jurisdictionId: string
+  roundId: string
+}
+
+const BatchRoundDataEntry: React.FC<IProps> = ({
+  electionId,
+  jurisdictionId,
+  roundId,
+}) => {
+  const batchesResp = useBatches(electionId, jurisdictionId, roundId)
+  const contests = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+  const recordBatchResults = useRecordBatchResults(
+    electionId,
+    jurisdictionId,
+    roundId
+  )
+  const { confirm, confirmProps } = useConfirm()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [selectedBatchId, setSelectedBatchId] = useState<IBatch['id']>()
```

Comment:
I wonder if we should auto-select the first batch. I think it might work a bit better with the feeling of "when I filter the batches, the top one in the list is selected"

Reply:
Good call: https://github.com/votingworks/arlo/pull/1644/commits/28bfee21a2ed884fdbe78b2be76fa80467bcbbf1